### PR TITLE
Chore: Remove pointer dependency

### PR DESCRIPTION
**Pull request by zserge**
Looks like we heavily use `github.com/xorcare/pointer` library which is nothing but a collection of value-to-pointer converters, like:

```go
func String(v string) *string { return &v }
```

Seems like it can be replaced with:

```go
func Pointer[T any](v T) *T { return &v }
```

Which is exactly what this PR does. One less dependency.

### CODE REVIEW

**Comment by itsmylife** [comment]
@zserge Removing a dependency is great but we already have this function here 
https://github.com/grafana/grafana/blob/main/pkg/kindsys/util.go
@sdboyer added this for [schemas](https://raintank-corp.slack.com/archives/C03BH67DS2J) and [kind system](https://github.com/grafana/grafana/blob/main/pkg/kindsys/EXTENDING.md). I think we can use it from there or apply this and drop it from there.

**Comment by zserge** [comment]
Oh, nice, thanks! Glad to see I'm not the only one thinking of this.
Yet, I'm more inclined to having all such utility functions in `pkg/util`, which serves a more general purpose and contains utils for strings, checksums, collections, contexts and now - pointers.
I would suggest `kindsys.Ptr` to be an alias to `util.Pointer` and/or use `util.Pointer` directly.

**Comment by sakjur** [comment]
I would suggest adding a blocker in depguard to discourage the reintroduction of the library?

https://github.com/grafana/grafana/blob/3abaf32cf26bc5dc4ecd54dd89c614cce6aebb83/.golangci.toml#L20-L22

**Review by papagian — APPROVED** [review]

**Comment by sdboyer** [comment]
i understand the reasoning there and agree that in the current context, an alias would make sense. However, it'll become a problem shortly, as we're getting ready to separate kindsys into a standalone repository, with one significant motivation being removing direct external dependencies on grafana/grafana.

tbh this is a one-line function with clear utility purposes. I've written it as an unexported function any number of times. i think the Go adage "a little copying is better than a little dependency" applies in this scenario.

**Review by DanCech — APPROVED** [review]
lgtm

File: .golangci.toml [hunk 1]
```diff
@@ -21,6 +21,8 @@ packages = ["io/ioutil"]
 "io/ioutil" = "Deprecated: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details."
 "gopkg.in/yaml.v2" = "Grafana packages are not allowed to depend on gopkg.in/yaml.v2 as gopkg.in/yaml.v3 is now available"
 "github.com/pkg/errors" = "Deprecated: Go 1.13 supports the functionality provided by pkg/errors in the standard library."
+"github.com/xorcare/pointer" = "Use pkg/util.Pointer instead, which is a generic one-liner alternative"
+>>>>>>> 87bc352f70 (deprecate xorcare/pointer library in linter)
```
Comment (sakjur) [inline .golangci.toml]:
```suggestion
```

**Review by sakjur — COMMENTED** [review]
FYI, there's one line left-over from resolving a merge conflict that needs to be deleted, then let's merge this 😄 :+1:

**Review by sakjur — APPROVED** [review]

**Pull request merged**